src/main/java/com/conveyal/datatools/manager/extensions/ExternalFeedResource.java

```diff
@@ -1,22 +1,233 @@
+     */
+    default void feedVersionCreated(FeedVersion feedVersion, String authHeader) {
+        LOG.info("feed version created! (no further action taken)");
+    }
+
+    /**
+     * Default method for checking whether an external feed source already exists in the project (using the
+     * getId and getIdKey interface methods).
+     * @param project
+     * @param externalFeed
+     * @return
+     */
+    default FeedSource checkForExistingFeed(Project project, ExternalFeed externalFeed) {
+        // check if a feed already exists with this id
+        for (FeedSource existingSource : project.getProjectFeedSources()) {
+            ExternalFeedSourceProperty idValue =
+                    ExternalFeedSourceProperty.find(existingSource, this.getResourceType(), externalFeed.getIdKey());
+            if (idValue != null && idValue.value != null && idValue.value.equals(externalFeed.getId())) {
+                return existingSource;
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Default method for getting JSON string from external feeds resource. If authHeader is provided/non-null,
+     * it will be used to authenticate against the server to which the request is made.
+     * @param url
+     * @param authHeader
+     * @return
+     */
+    default String getFeedsJson (URL url, String authHeader) {
+        if (url == null) {
+            LOG.warn("URL must not be null");
+            return null;
+        }
+
+        StringBuilder response = new StringBuilder();
+        try {
+            HttpURLConnection con = (HttpURLConnection) url.openConnection();
+
+            // optional default is GET
+            con.setRequestMethod("GET");
+
+            //add request header
+            con.setRequestProperty("User-Agent", "User-Agent");
+
+            if (authHeader != null) {
+                // add auth header
+                LOG.info("authHeader=" + authHeader);
+                con.setRequestProperty("Authorization", authHeader);
+            }
+
+
+            LOG.info("\nSending 'GET' request to URL : {}", url);
+            LOG.info("Response: {}", con.getResponseCode());
+
+            BufferedReader in = new BufferedReader(new InputStreamReader(con.getInputStream()));
+            String inputLine;
+
+            while ((inputLine = in.readLine()) != null) {
+                response.append(inputLine);
+            }
+            in.close();
+        } catch (IOException ex) {
+            LOG.error("Could not read from {} API", getResourceType());
+            ex.printStackTrace();
+            return null;
+        }
+
+        return response.toString();
+    }
+
+    /**
+     * Default method for importing an external feed resource into a project.
+     * @param project
+     * @param externalFeed
+     */
+    default void importFeed(Project project, ExternalFeed externalFeed) {
```

Comment:
If I understand correctly, we're putting the functionality that records the new feed into the plugin module. There is a default method that performs this action but it can be overridden. This means anyone who writes an extension must be familiar with our internals and be trusted not to trash internal state.

It seems to me that the more modular/safe/standard way to do this is have a single method within the core datatools project that handles importing a feed (which cannot be overridden), and have extension modules that only provide the core system with a list of objects representing candidates for import/update.